src/config.rs

```diff
@@ -119,4 +119,14 @@ fn get_remote(content: &String) -> (String, u16) {
         })
         .next()
         .unwrap();
+
+    return content
+        .lines()
+        .filter(|p| p.starts_with("proto "))
+        .map(|p| {
+            let proto = (&p[p.rfind(" ").unwrap() + 1..]).parse::<String>().unwrap().to_string();
+            (map.0.clone(), map.1, proto)
+        })
+        .next()
+        .unwrap();
```

Comment:
I know that this is basically the same as the code above, but we can simplify with [`strip_prefix`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix) and a `filter_map` instead of `filter` then `map`, and leave the udp default just in case:

```suggestion
    let proto = content
        .lines()
        .filter_map(|p| p.strip_prefix("proto "))
        .next()
        .unwrap_or("udp")
        .to_string();

    return (remote.0.to_string(), remote.1.to_string(), proto)
```